pkg/splunk/enterprise/util.go

```diff
@@ -2076,22 +2076,21 @@ func isAppFrameworkMigrationNeeded(afwStatusContext *enterpriseApi.AppDeployment
 	return afwStatusContext != nil && afwStatusContext.Version < currentAfwVersion && len(afwStatusContext.AppsSrcDeployStatus) > 0
 }
 
-// updateCRStatus fetches the latest CR, and on top of that, updates latest status
-func updateCRStatus(ctx context.Context, client splcommon.ControllerClient, origCR splcommon.MetaObject) {
+// updateCRStatus fetches the latest CR, and on top of that, updates latest status including error messages as well
+func updateCRStatus(ctx context.Context, client splcommon.ControllerClient, origCR splcommon.MetaObject, crError *error) {
```

Comment:
I do not think you need error pointer receiver to be passed here , you are not manipulating the error message. its just for read only.

Reply:
Per this [comment](https://github.com/splunk/splunk-operator/pull/1335#pullrequestreview-2113664379), closing this.